ios/FluentUI/Core/Theme/FluentTheme.swift

```diff
@@ -92,19 +92,33 @@ public class FluentTheme: NSObject, ObservableObject {
     @objc(sharedTheme)
     public internal(set) static var shared: FluentTheme = FluentThemeKey.defaultValue {
         didSet {
-            UIApplication.shared.connectedScenes
-                .compactMap {
-                    $0 as? UIWindowScene
-                }
-                .flatMap {
-                    $0.windows
-                }
-                .forEach { window in
-                    NotificationCenter.default.post(name: .didChangeTheme, object: window)
-                }
+            NotificationCenter.default.post(name: .didChangeTheme, object: nil)
         }
     }
 
+    /// Determines if a given `Notification` should cause an update for the given `UIView`.
+    ///
+    /// - Parameter notification: A `Notification` object that may be requesting a view update based on a theme change.
+    /// - Parameter view: The `UIView` instance that wants to determine whether to update.
+    ///
+    /// - Returns: `True` if the view should update, `false` otherwise.
+    @objc(isApplicableThemeChangeNotification:forView:)
+    public static func isApplicableThemeChange(_ notification: Notification,
+                                               for view: UIView) -> Bool {
+        // Do not update unless the notification's name is `.didChangeTheme`.
+        guard notification.name == .didChangeTheme else {
```

Comment:
and our swift ui views are still relying that hostviews are listening into it? do swiftui environment trait still work as expected?

Reply:
Yes! The code in `ControlHostingView` bridges between this notification code (for when people set the theme directly on a `UIView`/`UIWindow`) and SwiftUI environment variables, where the theme is set using the `.fluentTheme(_:)` view modifier.